Tasks/AzureWebPowerShellDeployment/Publish-AzureWebDeployment.ps1

```diff
@@ -95,14 +93,67 @@ if($WebSiteLocation)
 $azureCommand = "Publish-AzureWebsiteProject"
 if ($Slot)
 {
-    $azureCommandArguments = "-Name `"$WebSiteName`" -Package `"$packageFile`" -Slot `"$Slot`" $AdditionalArguments"
+    $azureCommandArguments = "-Name `"$WebSiteName`" -Package `"$packageFile`" -Slot `"$Slot`" $AdditionalArguments -ErrorVariable publishAzureWebsiteError"
 }
 else
 {
-    $azureCommandArguments = "-Name `"$WebSiteName`" -Package `"$packageFile`" $AdditionalArguments"
+    $azureCommandArguments = "-Name `"$WebSiteName`" -Package `"$packageFile`" $AdditionalArguments -ErrorVariable publishAzureWebSiteError"
 }
 $finalCommand = "$azureCommand $azureCommandArguments"
 Write-Host "$finalCommand"
 Invoke-Expression -Command $finalCommand
 
+#Update Deployment status - https://github.com/projectkudu/kudu/wiki/REST-API#deployment
+
+if($azureWebSite) {
+    
+    $status = 3 #failed
+    if(!$publishAzureWebsiteError) {
+        $status = 4 #succeeded
+    }
+
+    $username = $azureWebSite.PublishingUsername
+    $securePwd = ConvertTo-SecureString $azureWebSite.PublishingPassword -AsPlainText -Force
+    $credential = New-Object System.Management.Automation.PSCredential ($username, $securePwd)
+    
+    $author = Get-TaskVariable $distributedTaskContext "build.sourceVersionAuthor"
```

Comment:
For AT/DT mismatch, consider handling this gracefully to short-circuit if this is required.